stdlib/public/core/IntegerTypes.swift.gyb

```diff
@@ -1195,9 +1195,11 @@ public struct ${Self}
     // undefined if it overflows.
 %     if not (FloatBits == 16 and bits >= 32): # Float16 is always in-range for 32- and 64-bit ints.
     guard source > ${str(lower)}.0 && source < ${str(upper)}.0 else {
+%     else:
```

Comment:
We also need to handle UInt16, I think (max value 65536, which rounds to infinity in Float16).

Reply:
Alternatively, I think we could probably simply eliminate the gyb conditional and use `source.isFinite && existing condition`, right?

Reply:
@stephentyrone That's already the case: 'UInt16' is excluded because the condition is 'bits >= 32' (you wrote it, not me!).